### PR TITLE
Adding NuGet check and install

### PR DESCRIPTION
**Pull request by larsiwer**

### CODE REVIEW

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 1]
```diff
@@ -42,12 +42,12 @@ $CommonFunc = {
     #
     # Show arbitrary normal status message, with optional color coding
     #
-    function Show-Update(
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Please re-merge - no changes here?

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 2]
```diff
@@ -731,7 +731,17 @@ function Get-SddcDiagnosticInfo
 
     if ($S2DEnabled -ne $true) {
         if ((Test-NetConnection -ComputerName 'www.microsoft.com' -Hops 1 -ErrorAction SilentlyContinue -WarningAction SilentlyContinue).PingSucceeded) {
-            Compare-ModuleVersion
+            # The update check requires the NuGet provider version 2.8.5.201 or greater
+            $NuGetProvider = Get-PackageProvider -Name NuGet | Where-Object -FilterScript {$_.Version -gt 2.8.5.201}
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: |? { } is idiomatic for where-object

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 3]
```diff
@@ -731,7 +731,17 @@ function Get-SddcDiagnosticInfo
...
+            If (-not $NuGetProvider) {
+                # Install NuGet provider if necessary -- this will surpress the prompt to install
+                Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force -ErrorAction SilentlyContinue | Out-Null
+                $NuGetProvider = Get-PackageProvider -Name NuGet | Where-Object -FilterScript {$_.Version -gt 2.8.5.201}
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: |? { }

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 1 above)
Comment (larsiwer) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: I have no idea how this got in here and why git believes that there are changes.

**Comment by larsiwer** [comment]
Changed Where-Object to ? as requested. 
Not sure how I can easily do the re-merge for the Show-Update function.

**Pull request merged**